File: scratch/test_upload.js
const axios = require('axios');
const FormData = require('form-data');

async function test() {
    const formData = new FormData();
    formData.append('key', 'test_key_' + Date.now());
    formData.append('encryptedContent', 'test_content');
    formData.append('iv', 'auto');
    formData.append('viewLimit', '1');
    formData.append('expiryValue', '1');
    formData.append('expiryUnit', 'Hours');
    formData.append('file', Buffer.from('test file content'), 'test.txt');

    try {
        const response = await axios.post('http://localhost:5000/api/secrets', formData, {
            headers: formData.getHeaders()
        });
        console.log('Status:', response.status);
        console.log('Data:', response.data);
    } catch (error) {
        if (error.response) {
            console.log('Status:', error.response.status);
            console.log('Data:', error.response.data);
        } else {
            console.log('Error:', error.message);
        }
    }
}

test();

File: scratch/test_upload.py
import requests

url = "http://localhost:5000/api/secrets"
data = {
    "key": "test_key_123",
    "encryptedContent": "test_encrypted_content",
    "iv": "auto",
    "viewLimit": "1",
    "expiryValue": "1",
    "expiryUnit": "Hours"
}

files = {
    'file': ('test.txt', 'this is a test file content')
}

try:
    response = requests.post(url, data=data, files=files)
    print(f"Status Code: {response.status_code}")
    print(f"Response Body: {response.text}")
except Exception as e:
    print(f"Error: {e}")
